text/068-coqide-split.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: Future of CoqIDE
+Drivers: Karl Palmskog (@palmskog), Théo Zimmermann (@Zimmi48)
+---
+
+## Summary
+
+The goal of this CEP is:
+
+1. to clarify the priority and focus shift by the Coq team away from maintaining and evolving CoqIDE;
+2. to give an opportunity to the community of CoqIDE users to assume maintenance  of and lead the future evolution of CoqIDE through a source code split out of the Coq GitHub repository.
+
+More specifically, the proposal is to make an open call for new volunteer maintainers of CoqIDE through Coq community channels, and then to proceed with a source code split only if volunteer maintainers are found.
+
+## Context and motivation
+
+[CoqIDE](https://coq.inria.fr/refman/practical-tools/coqide.html) is an Integrated Development Environment (IDE) for Coq. CoqIDE is implemented using the OCaml programming language and the GTK3 widget toolkit for Graphical User Interfaces (GUIs), thanks to the [lablgtk3](https://opam.ocaml.org/packages/lablgtk3/) OCaml package.
+
+CoqIDE communicates with Coq thanks to an [XML protocol](https://github.com/coq/coq/blob/master/dev/doc/xml-protocol.md), implemented by the coqidetop server (in the coqide-server opam package), which itself is based on a component in Coq called the [State Transaction Machine](https://coq.github.io/doc/master/api/coq-core/Stm/index.html) (STM).
+
+The drawbacks of CoqIDE and coqidetop are that:
+
+- maintaining a standalone IDE just for Coq means that a lot of standard work has to be redone and lots of standard IDE features are not implemented,
+- the XML protocol is entirely custom and suited for the CoqIDE implementation, which means that any IDE wanting to interface with Coq through this protocol has to reimplement a lot of things, and that it won't always be well-suited to the model that the IDE expects.
```

Comment:
I think it would be useful to quantify the "lot" in "a lot of standard work has to be redone", in "lots of standard IDE features are not implemented", and in "any IDE wanting to interface with Coq through this protocol has to reimplement a lot of things".

In particular, is it only my perception or not, but I don't see LSP as magically solving the question of implementing features, or compliance to a protocol. If my perception is correct, it is probably worth to be made clear.

Reply:
It would require a proper evaluation, which I'm not able to conduct myself, but which I would be interested in if it were to happen.

Reply:
Indeed I think my comment above reflected the same thing, LSP solves a lot of practical problems (especially when bootstrapping), but from the Coq side it has little importance, in the sense that what we need to agree upon is _how_ we want to implement something such as a LSP server.